lib/src/system_cache.dart

```diff
@@ -62,20 +60,20 @@ class SystemCache {
   final sources = SourceRegistry();
 
   /// The sources bound to this cache.
-  final _boundSources = <Source, BoundSource>{};
+  final _boundSources = <Source?, BoundSource>{};
 
   /// The built-in Git source bound to this cache.
-  BoundGitSource get git => _boundSources[sources.git] as BoundGitSource;
+  BoundGitSource? get git => _boundSources[sources.git] as BoundGitSource?;
```

Comment:
Should never be null
```suggestion
  BoundGitSource get git => _boundSources[sources.git] as BoundGitSource;
```